src/main.c

```diff
@@ -679,6 +732,37 @@ poe_stream_open(char *dev, struct ustream_fd *s, speed_t speed)
 	return 0;
 }
 
+static int
+i2c_addr_open(int i2c_bus, char *filename, size_t size)
```

Comment:
As with the other reply, I know `main.c` is currently already messy with medium-specific implementations, but I'd prefer we separated media-handling out. We're not trying to clone `poemgr`, but this is how `poemgr` does it: note that `poemgr.c` contains no media-handling-specific headers, while the specific [PD IC file `pd69104.c`](https://github.com/blocktrron/poemgr/blob/master/pd69104.c) does.